api/scripts/prodtopreprod.js

```diff
@@ -0,0 +1,212 @@
+const fs = require("fs");
+const dotenv = require("dotenv");
+const mongoose = require("mongoose");
+const faker = require("faker");
+
+const { Client } = require("@elastic/elasticsearch");
+
+const envProdConfig = dotenv.parse(fs.readFileSync("../.env-prod"));
+const envStagingConfig = dotenv.parse(fs.readFileSync("../.env-staging"));
+
+const DB_ENDPOINT_PRODUCTION = envProdConfig.MONGO_URL;
+const ES_ENDPOINT_PRODUCTION = envProdConfig.ES_ENDPOINT;
+
+const DB_ENDPOINT_STAGING = envStagingConfig.MONGO_URL;
+const ES_ENDPOINT_STAGING = envStagingConfig.ES_ENDPOINT;
+
+(async () => {
+  await migrate("young", (doc) => {
+    if (doc.email === "se.legoff@gmail.com") return doc;
+
+    doc.set({ email: `${doc._id}@mail.com` });
+    doc.set({ firstName: faker.name.firstName() });
+    doc.set({ lastName: faker.name.lastName() });
+    return doc;
+  });
+  await migrate("referent", (doc) => {
+    if (doc.email === "se.legoff@gmail.com") return doc;
+
+    doc.set({ email: `${doc._id}@mail.com` });
+    doc.set({ firstName: faker.name.firstName() });
+    doc.set({ lastName: faker.name.lastName() });
+    return doc;
+  });
+  console.log("END");
+})();
+
+async function migrate(model, cb) {
+  const connection_mongo_prod = await createConnection(DB_ENDPOINT_PRODUCTION);
+  const connection_mongo_staging = await createConnection(DB_ENDPOINT_STAGING);
+
+  const connection_es_prod = new Client({ node: `https://${ES_ENDPOINT_PRODUCTION}` });
+  const connection_es_staging = new Client({ node: `https://${ES_ENDPOINT_STAGING}` });
+
+  let start_at = Date.now();
+
+  // TO prevent all midlleware (Sendinblue etc to work)
+  const Schema = require(`../src/models/${model}`).schema;
+  const ModelProduction = connection_mongo_prod.model(model, Schema);
+  const ModelStaging = connection_mongo_staging.model(model, Schema);
+
+  //clean staging
+  console.log(`Start delete staging db ${model}`);
+  await ModelStaging.deleteMany({});
+  console.log(`End delete staging db  ${model}`);
+
+  console.log(`Start delete staging es ${model}`);
+  try {
+    const exists = await connection_es_staging.indices.exists({ index: model });
+    if (exists) await connection_es_staging.indices.delete({ index: model });
+  } catch (e) {
+    console.log("e");
```

Comment:
On aura pas grand chose avec "e"

```suggestion
    console.log(e);
```